Fix resetting of entry editor prefs

HamzehAdawi opened the pull request:
Closes #14401 

This PR fixes the "Enable resetting of EntryEditorPreferences" issue.

### Steps to test

### Mandatory checks

- [x] I own the copyright of the code submitted and I license it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] I manually tested my changes in running JabRef (always required)
- [/] I added JUnit tests for changes (if applicable)
- [/] I added screenshots in the PR description (if change is visible to the user)
- [/] I described the change in `CHANGELOG.md` in a way that is understandable for the average user (if change is visible to the user)
- [/] I checked the [user documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request updating file(s) in <https://github.com/JabRef/user-documentation/tree/main/en>.

koppor reviewed (changes requested):
Which part of the following image didn't you understand?

<img width="961" height="695" alt="grafik" src="https://github.com/user-attachments/assets/13aa906b-4252-41b6-8b4d-33f862eb71a5" />

HamzehAdawi commented:
> Which part of the following image didn't you understand?
> <img alt="grafik" width="961" height="695" src="https://private-user-images.githubusercontent.com/1366654/518843187-13aa906b-4252-41b6-8b4d-33f862eb71a5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQxMDU3NzcsIm5iZiI6MTc2NDEwNTQ3NywicGF0aCI6Ii8xMzY2NjU0LzUxODg0MzE4Ny0xM2FhOTA2Yi00MjUyLTQxYjYtOGI0ZC0zM2Y4NjJlYjcxYTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTEyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMjVUMjExNzU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTRlNWMyMzFmZmMwODU0ZWQ5OTAwOTVhZjU2OWMyYzQ0MWUyNzhmNzE4M2QzMjg5YjM0MmJkM2MyYTUzMGI1NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.vthY-6Fb9rc4Vj0XxCzlss3KzLeB3cLUDGK5kIaP5O0">

@koppor sorry if I am misunderstanding, I could not find any defaults for EntryEditorPreferences besides the two I had already replaced. Am I missing something?

calixtus commented on jabgui/src/main/java/org/jabref/gui/entryeditor/EntryEditorPreferences.java:
```diff
@@ -274,4 +320,12 @@ public DoubleProperty previewWidthDividerPositionProperty() {
+
+    public double getPreviewDividerPos() {
+        return previewWidthDividerPosition.get();
+    }
```
mh... looks almost the same. I guess the AI did not catch that.

calixtus commented on jabgui/src/main/java/org/jabref/gui/entryeditor/EntryEditorPreferences.java:
```diff
@@ -274,4 +320,12 @@ public DoubleProperty previewWidthDividerPositionProperty() {
+
+    public String getJournalPopUp() {
+        return enablementStatus.get().toString();
+    }
```
should?

calixtus commented:
Try to use less AI and check your commits before pushing them. There are really sloppy mistakes that could be easily avoided by reading the code instead of just running an agent. Open source is not about writing the most code, but understand it and maybe also learn a little bit.

Please fix the mistakes.

HamzehAdawi commented:
Removed the redundant methods and fixed EntryEditorPreferences, as requested. Please let me know if any additional changes are needed, sorry for the oversight.

calixtus commented on jablib/src/main/java/org/jabref/logic/preferences/JabRefCliPreferences.java:
```diff
@@ -698,9 +687,7 @@ public JabRefCliPreferences() {
         defaults.put(AUTOLINK_EXACT_KEY_ONLY, Boolean.FALSE);
-        defaults.put(AUTOLINK_FILES_ENABLED, Boolean.TRUE);
         defaults.put(LOCAL_AUTO_SAVE, Boolean.FALSE);
-        defaults.put(ALLOW_INTEGER_EDITION_BIBTEX, Boolean.FALSE);
```
You found an artifact in the code we overlooked when we separated JabRefCli and JabRefGui. The constants about the EntryEditorPreferences called to in the put method here should not reside in JabRefCliPreferences class, but in the JabRefGuiPreferences class. Can you please move them? Thank you.

HamzehAdawi commented:
Moved EntryEditorPref constants from JabRefCli to JabRefGui. 

@calixtus regarding the EntryEditorTabs, I am not entirely sure if this is what you had in mind, so please let me know if anything should be adjusted there.

calixtus commented on jabgui/src/main/java/org/jabref/gui/preferences/JabRefGuiPreferences.java:
I think that this does not work. defaults map will be removed in the future. At least CUSTOM_TAB_NAME and CUSTOM_TAB_FIELDS must be removed from defaults map.

Editor tabs will be appended with this code to the default ones.

Did you test your changes or just push it? You said so when checking the mandatory checks.

calixtus commented on jabgui/src/main/java/org/jabref/gui/entryeditor/EntryEditorPreferences.java:
```diff
@@ -88,6 +89,51 @@ public EntryEditorPreferences(Map<String, Set<Field>> entryEditorTabList,
+    private EntryEditorPreferences() {
+        this(
+                Collections.emptyMap(),              // Default entryEditorTabList
+                Collections.emptyMap(),              // Default defaultEntryEditorTabList
```
These methods need to be adapted:

<img width="1327" height="1440" alt="Image" src="https://github.com/user-attachments/assets/37ad90e0-7114-4ec2-a35c-94b72e76b647" />

calixtus replied:
This one also:

<img width="1242" height="653" alt="Image" src="https://github.com/user-attachments/assets/e0fbdc1a-eb5f-4618-ab8b-7ac864c28356" />
